### PR TITLE
[MRG] ENH: add average_dipoles function to Dipole

### PR DESCRIPTION
**Pull request by blakecaldwell**
Split from #83 for a fast merge

### CODE REVIEW

**Comment by jasmainak** [comment]
what do you think of adding a parameter `nave` that is `1` by default. If it's not equal to 1, then we raise an error in `average_dipole` and a warning in `write`.

File: hnn_core/dipole.py [hunk 1]
```diff
@@ -69,6 +69,40 @@ def read_dipole(fname, units='nAm'):
     return dpl
 
 
+def average_dipoles(dpls):
+    """Compute dipole averages over a list of Dipole objects. 'L2', 'L5' and 'agg'
+    components are averaged separately.
```
Comment (jasmainak) [inline hnn_core/dipole.py:74]: nitpick, should be single line

File: hnn_core/dipole.py [hunk 2]
```diff
@@ -69,6 +69,40 @@ def read_dipole(fname, units='nAm'):
...
+        'L2', 'L5' and 'agg' components
+
+    Returns
+    -------
+    dpl: instance of Dipole
+        A new dipole object with each component of `dpl.data` representing the
+        average over the same components in the input list
+    """
+    # need at least one Dipole to get times
+    if (len(dpls) < 2):
```
Comment (jasmainak) [inline hnn_core/dipole.py:89]: nitpick, paranthesis not needed.

**Comment by jasmainak** [comment]
Otherwise looks good. Let me know what you think of the `nave` or `n_averages` attribute

**Comment by blakecaldwell** [comment]
> Otherwise looks good. Let me know what you think of the `nave` or `n_averages` attribute

Does it get incremented by `average_dipole`? In #83 you said "update", but I can only see utility if it is incremented.

**Comment by jasmainak** [comment]
If you averaged 5 dipoles, I would set `nave` to 5 in the returned Dipole object. So yeah it's incremented, don't see a situation where it would decrement.

**Comment by blakecaldwell** [comment]
> If you averaged 5 dipoles, I would set `nave` to 5 in the returned Dipole object. So yeah it's incremented, don't see a situation where it would decrement.



> Otherwise looks good. Let me know what you think of the `nave` or `n_averages` attribute

I think it's a good addition. Although I didn't add it to savetxt. Creating an issue for later...

File: doc/whats_new.rst [hunk 3]
```diff
@@ -47,6 +47,8 @@ API
 
 - Make a context manager for parallel backends (JoblibBackend, MPIBackend), by `Blake Caldwell`_ in `#79 <https://github.com/jonescompneurolab/hnn-core/pull/79>`_
 
+- Add average_dipoles function to `hnn_core.dipole`, by `Blake Caldwell`_ in `#156 <https://github.com/jonescompneurolab/hnn-core/pull/156>`_
```
Comment (jasmainak) [inline doc/whats_new.rst:50]: you can make the function clickable here by doing :func:`hnn_core.average_dipoles`.

**Comment by rythorpe** [comment]
Looks good to me. Thanks @blakecaldwell!

**Comment by jasmainak** [comment]
I'm going to merge this. The `whats_new` issue can be fixed in another PR, not urgent. Thanks @blakecaldwell !

**Pull request merged**